### PR TITLE
FEATURE: initiate recheck for device issues from user alert PM.

### PR DESCRIPTION
**Pull request by vinothkannans**
Now we can initiate the issue recheck process from Discourse itself using the button in the user alert PM's issue list.

### CODE REVIEW

File: assets/javascripts/discourse/initializers/extend-for-kolide.js [hunk 1]
```diff
@@ -7,12 +7,17 @@ import bootbox from "bootbox";
 function initializeWithApi(api) {
   const currentUser = api.getCurrentUser();
 
-  function attachAssignButton(cooked) {
-    const buttons = cooked.querySelectorAll("a.kolide-assign") || [];
+  function attachButtons(cooked) {
```
Comment (nbianca) [inline assets/javascripts/discourse/initializers/extend-for-kolide.js:10]: This does not really attach any buttons, but `registerButtonEvents` or something like that.

File: assets/javascripts/lib/discourse-markdown/kolide-recheck.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  buffer.push(token);
+
+  token = new state.Token("text", "", 0);
+  token.content = I18n.t("discourse_kolide.button.recheck");
+  buffer.push(token);
+
+  token = new state.Token("a_close", "a", -1);
+  buffer.push(token);
+
+  return;
```
Comment (nbianca) [inline assets/javascripts/lib/discourse-markdown/kolide-recheck.js:26]: This does not do anything.

File: app/models/kolide/issue.rb [hunk 3]
```diff
@@ -7,6 +7,14 @@ class Issue < ::ActiveRecord::Base
     belongs_to :device
     belongs_to :check
 
+    def identifier
```
Comment (nbianca) [inline app/models/kolide/issue.rb:10]: Maybe this should be named `id_md` or `identifier_markdown` or something like that.

**Review by nbianca — APPROVED** [review]
Just a couple of comments, but otherwise it looks good.

File: assets/javascripts/discourse/initializers/extend-for-kolide.js (hunk 1 above)
Comment (vinothkannans) [reply assets/javascripts/discourse/initializers/extend-for-kolide.js:10]: No, I'm using the `decorateCookedElement` method to create buttons inside the post content.

File: assets/javascripts/lib/discourse-markdown/kolide-recheck.js (hunk 2 above)
Comment (vinothkannans) [reply assets/javascripts/lib/discourse-markdown/kolide-recheck.js:26]: I'm using the hyperlink (`<a>`) tag as a button. If I didn't return false, the page will navigate to the top (`href="#"`).

**Pull request merged**